### PR TITLE
Implement enum to enum conversion when using to-enum operator

### PR DESCRIPTION
**Pull request by davidalk**

### CODE REVIEW

File: rosetta-integration-tests/src/test/java/com/regnosys/rosetta/generator/java/expression/ToEnumOperationTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                """).compile();
+        
+        Enum<?> result = model.evaluateExpression(Enum.class, """
+                    Bar -> VALUE2 to-enum Foo
+                    """);
+        
+        var expected = model.getEnumJavaValue("Foo", "VALUE2");
+        
+        assertEquals(expected, result);
+    }
```
Comment (SimonCockx) [inline rosetta-integration-tests/src/test/java/com/regnosys/rosetta/generator/java/expression/ToEnumOperationTest.java:63]: Perhaps also add a test that it handles empty values well?

E.g., `(if False then Bar -> VALUE2) to-enum Foo`

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/expression/ExpressionGenerator.xtend [hunk 2]
```diff
@@ -1083,14 +1083,17 @@ class ExpressionGenerator extends RosettaExpressionSwitch<JavaStatementBuilder,
...
+		val argumentJavaType = typeProvider.getRMetaAnnotatedType(expr.argument).RType.toJavaReferenceType
+		expr.argument.javaCode(MAPPER_S.wrapExtends(argumentJavaType), context.scope)
 			.collapseToSingleExpression(context.scope)
 			.mapExpression[JavaExpression.from('''«it».checkedMap("«expr.operator»", «conversion», «errorClass».class)''', MAPPER_S.wrap(expr))]
 	}
 
 	override protected caseToEnumOperation(ToEnumOperation expr, Context context) {
 		val javaEnum = expr.enumeration.buildREnumType.toJavaType
-		conversionOperation(expr, context, '''«javaEnum»::fromDisplayName''', IllegalArgumentException)
+		val argIsEnum = typeProvider.getRMetaAnnotatedType(expr.argument).RType instanceof REnumType
```
Comment (SimonCockx) [inline rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/expression/ExpressionGenerator.xtend:1094]: Just to be a bit more robust, you could add a `stripFromTypeAliases` here.

File: rosetta-lang/src/main/java/com/regnosys/rosetta/validation/ParseOperationValidator.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import org.eclipse.xtext.validation.Check;
+
+import com.regnosys.rosetta.RosettaEcoreUtil;
+import com.regnosys.rosetta.rosetta.expression.ParseOperation;
+import com.regnosys.rosetta.rosetta.expression.ToEnumOperation;
+import com.regnosys.rosetta.types.builtin.RBuiltinTypeService;
+
+import static com.regnosys.rosetta.rosetta.expression.ExpressionPackage.Literals.*;
+
+public class ParseOperationValidator extends AbstractDeclarativeRosettaValidator {
```
Comment (SimonCockx) [inline rosetta-lang/src/main/java/com/regnosys/rosetta/validation/ParseOperationValidator.java:15]: This one should be in the `validation.expression` package and reuse the same utilities as other expression checks for consistent validation messages.

Comment (SimonCockx) [reply rosetta-lang/src/main/java/com/regnosys/rosetta/validation/ParseOperationValidator.java:15]: See existing validation checks for `ParseOperation`s. It might be interesting to look at the `SwitchOperation` validator, which is also overloaded for multiple argument types.

File: rosetta-integration-tests/src/test/java/com/regnosys/rosetta/generator/java/expression/ToEnumOperationTest.java (hunk 1 above)
Comment (davidalk) [reply rosetta-integration-tests/src/test/java/com/regnosys/rosetta/generator/java/expression/ToEnumOperationTest.java:63]: Done, but I think the Mapper code handles this anyway. I had a look to see if I needed to do a null check in my conversion. Anyway the test is added.

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/expression/ExpressionGenerator.xtend (hunk 2 above)
Comment (davidalk) [reply rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/expression/ExpressionGenerator.xtend]: done

File: rosetta-lang/src/main/java/com/regnosys/rosetta/validation/ParseOperationValidator.java (hunk 3 above)
Comment (davidalk) [reply rosetta-lang/src/main/java/com/regnosys/rosetta/validation/ParseOperationValidator.java]: done

File: rosetta-integration-tests/src/test/java/com/regnosys/rosetta/validation/ExpressionValidatorTest.java [hunk 4]
```diff
@@ -18,119 +17,153 @@
...
+        RosettaExpression expr = modelService.toTestModel("""
+                 enum Bar:
+                     VALUE1
+                     VALUE2
+                """).parseExpression("""
+                123 to-enum Bar
+                """);
+
+        validationTestHelper.assertError(expr, TO_ENUM_OPERATION, null,
+                "Operator `to-enum` is not supported for type `int(digits: 3, min: 123, max: 123)`. Supported argument types are strings and enumerations.");
```
Comment (SimonCockx) [inline rosetta-integration-tests/src/test/java/com/regnosys/rosetta/validation/ExpressionValidatorTest.java:66]: Whoops, this is interesting!

Comment (davidalk) [reply rosetta-integration-tests/src/test/java/com/regnosys/rosetta/validation/ExpressionValidatorTest.java]: test util updated

**Review by SimonCockx — APPROVED** [review]
